betterjails/src/main/java/io/github/emilyydev/betterjails/data/PrisonerDataHandler.java

```diff
@@ -138,16 +140,25 @@ public void addJailedPlayer(
       knownLastLocation = existingPrisoner.lastLocationNullable();
     }
 
-    if (isPlayerOnline) {
+    // this is ugly
+    final Runnable teleportAction;
```

Comment:
making this not final removes the need for the newly added if cases

Reply:
fixed in https://github.com/emilyy-dev/BetterJails/pull/40/commits/8631c3f112b5adc19802766c288f48b8bc3fd775